### ISSUE TITLE
Enable Mutual SSL Authentication

### ISSUE THREAD

**Issue by atgjack** [googleapis/google-http-java-client#904]
It seems that the current implentations for SSL request do not support Mutual SSL authentication. Current outgoing requests do not use the `javax.net.ssl.keyStore` when making requests, which prevents the 2-way ssl handshake from going through.

The ~~[Apache ApacheHttpTransport](https://github.com/googleapis/google-http-java-client/blob/master/google-http-client/src/main/java/com/google/api/client/http/apache/ApacheHttpTransport.java#L151)~~ (UPDATE: this should really have been referred to v2 `ApacheHttpTransport`, although the same question applies to v1. see https://github.com/googleapis/google-http-java-client/issues/904#issuecomment-564625366) not using `getSystemSocketFactory`. That version takes into account the `javax.net.ssl` properties, and would most likely allow for the 2way ssl connection to complete. Is there a reason this is not being used?

**Comment by chanseokoh** [comment]
At least it's possible to override the `SSLSocketFactory` to whatever you like on the client side, if it helps at all. Some bad example (taken from [Jib code](https://github.com/GoogleContainerTools/jib/blob/3688f567835130eeca7c13b90a5023e485690bf9/jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java#L107)) to make an insecure connection:

```java
          ApacheHttpTransport.newDefaultHttpClientBuilder()
              .setSSLSocketFactory(null) // creates new factory with the SSLContext given below
              .setSSLContext(SslUtils.trustAllSSLContext())
              .setSSLHostnameVerifier(new NoopHostnameVerifier());
```

**Comment by elharo** [comment]
ApacheHttpTransport is deprecated and scheduled to be removed in about a year.

**Comment by chanseokoh** [comment]
I have some context. This is when using [Jib](https://github.com/GoogleContainerTools/jib), which uses the new v2 `ApacheHttpTransport` (`com.google.api.client.http.apache.v2.ApacheHttpTransport`). @atgjack should have referred to the v2 instead of the deprecated v1. I think @atgjack just searched the class at HEAD instead of searching on the v2 branch.

**Comment by chanseokoh** [comment]
In any case, I think the question whether it should have used `getSystemSocketFactory` applies generally regardless of v1 or v2. Although I haven't looked into the code, but if what @atgjack said is right, perhaps this library should use the system socket factory?

**Comment by chingor13** [comment]
This should be available in 1.38.0
